packages/react-compose/src/resolveSlotProps.ts

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import { OptionsResolverResult, defaultMappedProps } from './createOptionsResolver';
+
+export const NullRender = () => null;
+
+/**
+ * Helper utility which takes in a classes array from compose options, resolves functions,
+ * merges them into a final result, and distributes classnames to slotProps within the given
+ * resolver result object.
+ */
+export function resolveSlotProps(result: OptionsResolverResult): OptionsResolverResult {
+  const { state, slots, slotProps } = result;
+
+  Object.keys(slots).forEach((slotName: string) => {
+    const slot = slots[slotName];
+    let slotProp = state[slotName];
+
+    if (slot && slotProp !== undefined && slotProp !== null) {
+      const slotPropType = typeof slotProp;
```

Comment:
Don't we always want slotProp to be an object? Is this a way fo doing shorthand?

Reply:
slotProp can be:

* a literal value
* JSX
* an object